Simplenote/Classes/UITextView+Simplenote.swift

```diff
@@ -13,16 +14,41 @@ extension UITextView {
         return selectedRange.length > 0
     }
 
-    /// Returns the Selected Text's bounds
+    /// Indicates if there's an ongoing Undo Operation in the Text Editor
     ///
-    @objc
-    var selectedBounds: CGRect {
-        guard selectedRange.length > 0 else {
-            return .zero
+    var isUndoingEditOP: Bool {
+        undoManager?.isUndoing == true
+    }
+
+    /// Indicates if the user is Editing an Interlink
+    ///
+    var isEditingInterlink: Bool {
+        interlinkKeywordAtSelectedLocation != nil
+    }
+}
+
+
+// MARK: - Interlinks
+//
+extension UITextView {
+
+    /// Returns the Interlinking Keyword at the current Location (if any)
+    ///
+    var interlinkKeywordAtSelectedLocation: (Range<String.Index>, Range<String.Index>, String)? {
+        guard let text = text else {
+            return nil
         }
 
-        return layoutManager.boundingRect(forGlyphRange: selectedRange, in: textContainer)
+        return text.indexFromLocation(selectedRange.location).flatMap { index in
+            text.interlinkKeyword(at: index)
+        }
```

Comment:
Maybe we can do something like:
```
let range = Range(selectedRange, in: text)
let location = range.lowerBound
text.interlinkKeyword(at: location)
```

Reply:
Sort of! `Range(selectedRange, in: text)` returns an optional. However, I love the idea of using an SDK API instead.

Thank you for the heads up!